### PR TITLE
qu-auto/SORQA-271 Test pseudonymization in immunization

### PR DESCRIPTION
**Pull request by hms-sgent**
Fixes #

### CODE REVIEW

**Comment by hms-sgent** [comment]
https://jenkins.sormas.netzlink.com/job/sormas-RunE2eTests-branch/1419/
https://jenkins.sormas.netzlink.com/job/sormas-RunE2eTests-branch/1420/
https://jenkins.sormas.netzlink.com/job/sormas-RunE2eTests-branch/1422/

**Review by pk-sgent — APPROVED** [review]

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/entities/services/ImmunizationService.java [hunk 1]
```diff
@@ -68,4 +68,14 @@ public Immunization buildGeneratedImmunizationWithMeansOfImmunizationFromCase(
         .meansOfImmunization(meansOfImmunization)
         .build();
   }
+
+  public Immunization buildImmunizationWithSpecificResponsibleLocation() {
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/entities/services/ImmunizationService.java:72]: let's parametrize this method to allow reuse it also for other combinations of region+district

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/EditCaseSteps.java [hunk 2]
```diff
@@ -1919,6 +1920,35 @@ public EditCaseSteps(
...
+          webDriverHelpers.clickOnWebElementBySelector(SAVE_POPUP_CONTENT);
+          webDriverHelpers.waitForPageLoadingSpinnerToDisappear(30);
+          webDriverHelpers.waitUntilIdentifiedElementIsPresent(EditImmunizationPage.UUID);
+        });
+
+    And(
+        "^I navigate to linked immunization on Edit case page$",
+        () -> {
+          webDriverHelpers.waitUntilIdentifiedElementIsPresent(REPORT_DATE_INPUT);
+          System.out.print(
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/EditCaseSteps.java:1944]: remove system out and change it with logger if needed

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/immunizations/CreateNewImmunizationSteps.java [hunk 3]
```diff
@@ -88,6 +88,17 @@ public CreateNewImmunizationSteps(
           selectResponsibleDistrict(immunization.getResponsibleDistrict());
           webDriverHelpers.clickOnWebElementBySelector(IMMUNIZATION_POPUP_SAVE_BUTTON);
         });
+
+    When(
+        "^I fill a new immunization form with specific responsible location$",
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/immunizations/CreateNewImmunizationSteps.java:93]: here where the parametrized method will go, I would add region and district values into method naming, so the code will be readable and understand also the step(need) from scenario

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/immunizations/EditImmunizationSteps.java [hunk 4]
```diff
@@ -344,6 +344,49 @@ public EditImmunizationSteps(
...
+          ComparisonHelper.compareEqualFieldsOfEntities(
+              collectedImmunization,
+              createdImmunization,
+              List.of("dateOfReport", "responsibleRegion", "responsibleDistrict"));
+        });
+    Then(
+        "^I check that Immunization data is displayed as read-only on Edit immunization page$",
+        () -> {
+          webDriverHelpers.waitForPageLoadingSpinnerToDisappear(30);
+          TimeUnit.SECONDS.sleep(15);
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/immunizations/EditImmunizationSteps.java:362]: why this long sleep?

**Review by rdutu-vg — CHANGES_REQUESTED** [review]

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/entities/services/ImmunizationService.java (hunk 1 above)
Comment (hms-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/entities/services/ImmunizationService.java:72]: ok, I'll parametrize it

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/EditCaseSteps.java (hunk 2 above)
Comment (hms-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/EditCaseSteps.java:1944]: I forgot to remove it. This is not needed and I'll remove it.

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/immunizations/EditImmunizationSteps.java (hunk 4 above)
Comment (hms-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/immunizations/EditImmunizationSteps.java:362]: Somethimes when the spinner dissapear it takes time for the elements to be clickable. But you're right 5 seconds should be enough.

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/entities/services/ImmunizationService.java (hunk 1 above)
Comment (hms-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/entities/services/ImmunizationService.java]: done

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/EditCaseSteps.java (hunk 2 above)
Comment (hms-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/EditCaseSteps.java]: done

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/immunizations/CreateNewImmunizationSteps.java (hunk 3 above)
Comment (hms-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/immunizations/CreateNewImmunizationSteps.java]: done

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/immunizations/EditImmunizationSteps.java (hunk 4 above)
Comment (hms-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/immunizations/EditImmunizationSteps.java]: done

**Review by rdutu-vg — APPROVED** [review]

**Pull request merged**